crates/sui-metrics-proxy/src/handlers.rs

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use axum::{
+    body::Body,
+    extract::State,
+    http::{header, HeaderMap, HeaderValue, Request, StatusCode},
+    response::IntoResponse,
+};
+use std::sync::Arc;
+use tokio::sync::mpsc::Sender;
+use tracing::error;
+
+use crate::channels::NodeMetric;
+
+/// Publish handler which receives metrics from nodes.  Nodes will call us at this endpoint
+/// and we relay them to the upstream tsdb
+///
+/// An mpsc is used within this handler so that we can immediately return an accept to calling nodes.
+/// Downstream processing failures may still result in metrics being dropped.
+pub async fn publish_metrics(
+    State(state): State<Arc<Sender<NodeMetric>>>,
+    headers: HeaderMap,
+    request: Request<Body>,
+) -> impl IntoResponse {
+    if let Some(reason) = validate_headers(&headers) {
+        return (
+            StatusCode::BAD_REQUEST,
+            [(header::CONNECTION, "close")],
+            reason,
+        );
+    }
+    let host: String = match headers.get(header::HOST).map(|x| x.to_string()) {
+        Some(host) => host,
+        None => "unknown".to_string(),
+    };
```

Comment:
Maybe put a TODO here to fill in with the onchain validator name later?

Reply:
That's the http host field, it won't be used for the lable.  The onchain validator will and i need to wire that up still.  We only use this host field for logging purposes (which i forgot to even use in the upstream processor).